src/fairseq2/checkpoint.py

```diff
@@ -204,13 +215,28 @@ def __init__(
             The keys in provided checkpoints whose values are replicated across
             all processes in the gang.
         """
-        self._gang = gang
+        self._root_gang = gang
+
+        self._dp_gang = gang
+
+        self._shard_suffix = ""
+
+        if dp_gang is not None and tp_gang is not None:
+            self._dp_rank = dp_gang
```

Comment:
maybe this is a typo ? We are assigning gang to rank

Reply:
Thanks a lot for catching it! You are right, a stupid bug from my end. Fixed it in #493.